proto/internal/temporal/server/api/persistence/v1/hsm.proto

```diff
@@ -161,3 +168,39 @@ message StateMachinePath {
     repeated StateMachineKey path = 1;
 }
 
+// StateMachineOperation represents a history record of operations performed on nodes in the state machine tree
+message StateMachineOperation {
```

Comment:
You should store tombstones in ExecutionInfo, other than that you don't need this structure to be persistent.

Reply:
Will do in Part 2 replication PR